awx/ui_next/src/screens/Credential/CredentialDetail/CredentialDetail.jsx

```diff
@@ -0,0 +1,172 @@
+  const [contentError, setContentError] = useState(null);
+  const [deletionError, setDeletionError] = useState(null);
+  const [hasContentLoading, setHasContentLoading] = useState(true);
+  const history = useHistory();
+
+  useEffect(() => {
+    (async () => {
+      setContentError(null);
+      setHasContentLoading(true);
+      try {
+        const {
+          data: { inputs: credentialTypeInputs },
+        } = await CredentialTypesAPI.readDetail(credential_type.id);
+
+        setFields(credentialTypeInputs.fields);
+      } catch (error) {
+        setContentError(error);
+      } finally {
+        setHasContentLoading(false);
+      }
+    })();
+  }, [credential_type.id]);
+
+  const handleDelete = async () => {
+    setHasContentLoading(true);
+
+    try {
+      await CredentialsAPI.destroy(credentialId);
+      history.push('/credentials');
+    } catch (error) {
+      setDeletionError(error);
+    }
+    setHasContentLoading(false);
+  };
+
+  const renderDetail = ({ id, label, type, secret }) => {
+    let detail;
+
+    if (type === 'boolean') {
+      detail = (
+        <Detail
+          key={id}
+          label={i18n._(t`Options`)}
+          value={<List>{inputs[id] && <ListItem>{label}</ListItem>}</List>}
+        />
+      );
+    } else if (secret === true) {
+      detail = <Detail key={id} label={label} value={i18n._(t`Encrypted`)} />;
+    } else {
+      detail = <Detail key={id} label={label} value={inputs[id]} />;
+    }
+
+    return detail;
+  };
+
+  if (hasContentLoading) {
+    return <ContentLoading />;
+  }
+
+  if (contentError) {
+    return <ContentError error={contentError} />;
+  }
+
+  return (
+    <CardBody>
+      <DetailList>
+        <Detail label={i18n._(t`Name`)} value={name} />
+        <Detail label={i18n._(t`Description`)} value={description} />
+        {organization && (
+          <Detail
+            label={i18n._(t`Organization`)}
+            value={
+              <Link to={`/organizations/${organization.id}/details`}>
+                {organization.name}
+              </Link>
+            }
+          />
+        )}
+        <Detail
+          label={i18n._(t`Credential Type`)}
+          value={
+            <Link to={`/credential_types/${credential_type.id}/details`}>
+              {credential_type.name}
+            </Link>
+          }
+        />
+
+        {fields.map(field => renderDetail(field))}
+
+        <UserDateDetail
+          label={i18n._(t`Created`)}
+          date={created}
+          user={created_by}
+        />
+        <UserDateDetail
+          label={i18n._(t`Last Modified`)}
+          date={modified}
+          user={modified_by}
+        />
+      </DetailList>
+      <CardActionsRow>
+        {user_capabilities.edit && (
+          <Button component={Link} to={`/credentials/${credentialId}/edit`}>
+            {i18n._(t`Edit`)}
+          </Button>
+        )}
+        {user_capabilities.delete && (
+          <DeleteButton
```

Comment:
This need a different delete modal, but I can see an issue has already been made to address this. Awesome!